### PR TITLE
move ecl L1, TECS, and data validator to PX4/Firmware

### PR DESCRIPTION
**Pull request by dagar**
The PX4/ecl subprojects TECS, L1 (position controller), and data validator libraries were never really fleshed out as standalone projects. They don't have testing, basic documentation, or even a mention in PX4/ecl. I'm not aware of any standalone usage of these outside of PX4/Firmware.

Keeping these in a submodule under active, but unrelated EKF development is a small burden with no benefit.

Let's treat these as integral pieces of PX4 and lower the barrier to incremental improvement.

ECL change https://github.com/PX4/ecl/pull/844.

### CODE REVIEW

File: src/lib/CMakeLists.txt [hunk 1]
```diff
@@ -58,8 +60,8 @@ add_subdirectory(perf)
 add_subdirectory(pid)
 add_subdirectory(rc)
 add_subdirectory(systemlib)
+add_subdirectory(tecs)
 add_subdirectory(terrain_estimation)
 add_subdirectory(tunes)
 add_subdirectory(version)
 add_subdirectory(weather_vane)
-add_subdirectory(mag_compensation)
```
Comment (sfuhrer) [inline src/lib/CMakeLists.txt:65]: This is not directly linked to the rest of this PR, right?

Comment (dagar) [reply src/lib/CMakeLists.txt:65]: I sorted the list.

**Comment by sfuhrer** [comment]
@dagar given that this PR doesn't introduce any issues in SITL and we haven't heard any opposing voice it should be good to bring in I guess?

**Comment by dagar** [comment]
Silence implies consent...
I'll merge the ecl side and bring in the update here.

**Pull request merged**